python/dataset_plotting.py

```diff
@@ -128,73 +152,79 @@ def plot_1d(input_data, logx=False, logy=False, logxy=False, bars=False):
 # standard image made of pixels is created.
 # If plot=False, then not plot is produced, instead the layout and Data.Value
 # variable are returned.
-def plot_image(input_data, contours=False, plot=True):
+def plot_image(input_data, axes=None, contours=False, plot=True):
 
     ndim = len(input_data.dimensions())
+    if axes is not None:
+        naxes = len(axes)
+    else:
+        naxes = 0
     # TODO: this currently allows for plot_image to be called with a 3D dataset
     # and plot=False, which would lead to an error. We should think of a better
     # way to protect against this.
-    if (ndim > 1) and ((ndim < 3) or ((ndim < 5) and not plot)):
+    if (ndim > 1) and (((ndim < 3) or ((ndim < 5) and not plot)) or (naxes == 2)):
 
         values = []
+        coords = []
         for var in input_data:
-            if var.is_data:
+            if var.is_coord:
+                coords.append(var)
+            elif var.is_data:
                 values.append(var)
 
         if len(values) > 1:
             raise RuntimeError("More than one Data.Value found! Please use e.g."
-                               " plot_image(dataset.subset(Data.Value, 'sample'))"
+                               " plot_image(dataset.subset('sample'))"
                                " to select only a single Value.")
 
-        xcoord = input_data[dimensionCoord(values[0].dimensions.labels[0])]
-        ycoord = input_data[dimensionCoord(values[0].dimensions.labels[1])]
+        if axes is None:
+            xcoord = input_data[dimensionCoord(values[0].dimensions.labels[0])]
+            ycoord = input_data[dimensionCoord(values[0].dimensions.labels[1])]
+        else:
+            x_not_found = True
+            y_not_found = True
+            for c in coords:
+                dimcoord = dimensionCoord(c.dimensions.labels[0])
+                if dimcoord == axes[0]:
+                    xcoord = input_data[dimcoord]
+                    x_not_found = False
+                if dimcoord == axes[1]:
+                    ycoord = input_data[dimcoord]
+                    y_not_found = False
```

Comment:
I am not sure I get what is happening here. Why not just

```python
xcoord = input_data[axes[0]]
ycoord = input_data[axex[1]]
```
?

Reply:
yes i don't know why i was messing around here. fixed it now